test/functional/ui/screen.lua

```diff
@@ -450,8 +450,15 @@ end
 function Screen:expect_unchanged(waittime_ms, ignore_attrs, request_cb)
   waittime_ms = waittime_ms and waittime_ms or 100
   -- Collect the current screen state.
-  self:sleep(waittime_ms, request_cb)
-  local kwargs = self:get_snapshot(nil, ignore_attrs)
+  local kwargs
+  local function notification_cb(method, args)
+    assert(method == 'redraw')
+    assert(self:_redraw(args))  -- did flush
+    kwargs = self:get_snapshot(nil, ignore_attrs)
+  end
+  -- self._session:request('nvim_eval', '1')
+  run_session(self._session, nil, notification_cb, nil, 0)
```

Comment:
should pass on `request_cb`, in case a request comes before the first notification.

Reply:
Ok.

1. Is this ever passed into `Screen:expect_unchanged` though?  (at least it is not used explicitly)
2. Does calling `self._session:request('nvim_eval', '1')` (aka `wait()`) before would make a difference?

Reply:
I guess we want to have `request_cb` for all the functions here which runs the event loop. It should be possible to interleave screen tests with any test that needs requests from nvim to the client (ideally non-"redraw" notification as well).

Reply:
Still not clear to me.
Trying `self._session:request('nvim_eval', '1')  -- "wait()"` now additionally.